tests/unit/runner/test_runner.py

```diff
@@ -24,3 +30,199 @@ def _some_datetime(offset: int = 0):
 def test_running_time_elapsed(time_started, max_running_time, expected):
     """Test the `_running_time_elapsed` function."""
     assert _running_time_elapsed(time_started, max_running_time) == expected
+
+
+@pytest.fixture
+def runner_config():
+    """Naive minimal example to satisfy Runner instantiation."""
+
+    return Config(
+        yagna=YagnaConfig(
+            subnet_tag="public",
+        ),
+        payment=PaymentConfig(
+            budget=1,
+            driver="erc20",
+            network="rinkeby",
+        ),
+    )
+
+
+@pytest.fixture
+def dapp_descriptor():
+    """Naive minimal example to satisfy Runner instantiation."""
+    return DappDescriptor(
+        payloads={},
+        nodes={},
+    )
+
+
+@pytest.fixture
+def runner(runner_config, dapp_descriptor, mocker):
+    """Mostly mocked out Runner instance."""
+
+    # TODO: How to mock Golem inside of Runner?
+    runner = Runner(
+        config=runner_config,
+        dapp=dapp_descriptor,
+    )
+
+    # Naive simplification of two nodes
+    mocker.patch.object(
+        runner.dapp,
+        "nodes",
+        {
+            "foo": None,
+            "bar": None,
+        },
+    )
+
+    return runner
+
+
+@pytest.mark.skip("needs properly mocked `yapapi.Golem`")
+async def test_runner_desired_state(runner):
+    """Test to check if desired app state is properly managed with app lifetime."""
+    # TODO: Any way to avoid using non-public api?
+    assert runner._desired_app_state == ServiceState.pending
+
+    await runner.start()
+
+    assert runner._desired_app_state == ServiceState.running
+
+    await runner.stop()
+
+    assert runner._desired_app_state == ServiceState.terminated
+
+
+async def test_runner_app_state_pending():
+    """Test app state reporting at initial state of Runner."""
+    dapp_node_count = 2
+    desired_app_state = ServiceState.pending
+    nodes_states: Dict[str, Dict[int, ServiceState]] = {}
+
+    assert (
```

Comment:
given the identical nature of this assert in all the `test_runner_app_state_x` functions, I'd unify them and just add `dapp_node_count` and `desired_app_state` as parameters. I don't insist, though